src/imagery/i.sar/i.sar.speckle/i.sar.speckle.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python3
+
+############################################################################
+#
+# MODULE:	i.sar.speckle
+#
+# AUTHOR:   Margherita Di Leo
+#
+# PURPOSE:	Speckle noise removal for synthetic aperture radar (SAR) images
+#
+# COPYRIGHT: (C) 2002-2019 by the GRASS Development Team
+#
+# 		This program is free software under the GNU General Public
+# 		License (>=v2). Read the file COPYING that comes with GRASS
+# 		for details.
+#
+# REFERENCES:
+#
+# Lee, J. S. (1986). Speckle suppression and analysis for synthetic aperture
+# radar images. Optical engineering, 25(5), 255636.
+#
+#############################################################################
+
+# %Module
+# % description: Remove speckle from SAR image
+# % keyword: imagery
+# % keyword: speckle
+# % keyword: sar
+# % keyword: radar
+# % overwrite: yes
+# %End
+# %option G_OPT_R_INPUT
+# % key: input
+# % description: Name of input image
+# % required: yes
+# %end
+# %option G_OPT_R_OUTPUT
+# % key: output
+# % description: Name of output image
+# % required: yes
+# %end
+# %option
+# % key: method
+# % description: Method for speckle removal
+# % options: lee,mean,gauss
+# % answer: lee
+# % required: yes
+# %end
+# %option
+# % key: size
+# % type: integer
+# % description: Size of neighborhood
+# % answer: 11
+# % required: yes
+# %end
+# %option
+# % key: std
+# % type: float
+# % description: Standard deviation value in case gaussian filter is selected
+# % answer: 1
+# % required: no
+# %end
+
+
+import os
+import grass.script as grass
+from grass.pygrass.modules.shortcuts import general as g
+from grass.pygrass.modules.shortcuts import raster as r
+
+
+def lee_filter(img, size, img_out):
+    pid = str(os.getpid())
+    img_mean = "tmp%s_img_mean" % pid
+    img_sqr = "tmp%s_img_sqr" % pid
+    img_sqr_mean = "tmp%s_img_sqr_mean" % pid
+    img_variance = "tmp%s_img_variance" % pid
+    img_weights = "tmp%s_img_weights" % pid
```

Comment:
Please see best practices for creating temporary maps:
https://github.com/wenzeslaus/foss4g-2022-developing-custom-grass-tools/blob/main/notebook_3_best_practices.ipynb (look for Temporary Maps section)
(This will eventually end up in official documentation...)

The idea is to create safe names, register them so that they get removed if something fails, and avoid accidental removal of maps created by different processes.